### ISSUE TITLE
How to run against 50+ accounts

### ISSUE THREAD

**Issue by TheFlexican** [duo-labs/cloudmapper#605]
Hi there,
Not an issue, but more a general wondering. How would you go about to run the auditor against 50 accounts? Issues I'm facing at the moment, that the auditor starts auditing all 50 accounts at the same time, resulting in memory errors. I could increase the mem/cpu, or modify run script to build in some logic to only run x amount accounts at a given time.
Your thoughts on this ?

Thank you!

**Comment by 0xdabbad00** [comment]
Hmm, I admittedly had not considered the impact of all the collections running at once.  I had only run it against two accounts at the same time to ensure that aspect worked.  I see the following options:
1. Bump up the memory.  Right now it uses 512MB of memory. https://github.com/duo-labs/cloudmapper/blob/c84742e4a8d3f9d7988549b5fde086303efaded5/auditor/lib/cloudmapperauditor-stack.js#L83

2. Avoid running so many collections in parallel.  The bash script that controls the collection is a little complicated.  It basically calls `collect` on every account as a background process and then waits for each of those processes to finish.   https://github.com/duo-labs/cloudmapper/blob/c84742e4a8d3f9d7988549b5fde086303efaded5/auditor/s3_bucket_files/run_cloudmapper.sh#L39

Beware that if you run them more serially, you could bump into the 3 hour timeout I put in for the ECS, so you could bump that up. https://github.com/duo-labs/cloudmapper/blob/c84742e4a8d3f9d7988549b5fde086303efaded5/auditor/resources/entrypoint.sh#L11

**Comment by TheFlexican** [comment]
Hi,
I went for option 2 and 3. 
For option two, i simply added 'wait $!' after the call to the collect function, for option three i adjusted the timeout to 8 hours. 

Wondering if this can be done in de the code for example within the cdk_app.yaml, an option to run all the audits in parallel(how many at same time) or in serial.

Thank you for the help!

**Comment by 0xdabbad00** [comment]
I've renamed this issue to keep it as a feature enhancement request to help with this situation, that will likely spin up say 5 collections at a time, wait for all to complete, then spin up another 5, etc.

Thinking about this more though, I worry that if you have one account with 90% of your company's stuff in it, that this solution will not help.  I believe that CloudMapper should collect all of the data for a certain resource in an account and then write it to a file, and free up that memory, but I've never tested.  Even if that is true, for a single resource type, if you have 1 million of them, and each takes up 1K for the data to be saved, then that will use 1GB of memory.  

Also, CloudMapper largely works by reading all of the data about an account into memory, especially for anything network related, such as finding which EC2s are publicly exposed.  Even if you mute these audit checks, CloudMapper will still check for them, it just won't report them.

So basically, I think I would need to better identify what part of CloudMapper is causing memory issues and try to attack that, which could potentially be hard enough to fix that for large environments we might just have to say CloudMapper can't scale to that, or we have a feature flag to disable some aspect of CloudMapper for very large environments.
